src/devices/tuya.ts

```diff
@@ -8734,6 +8734,66 @@ const definitions: Definition[] = [
             ],
         },
     },
+    {
+        fingerprint: tuya.fingerprint('NAS-WV03B', ['_TZE204_rzrrjkz2']),
+        model: 'NAS-WV03B',
+        vendor: 'NEO',
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
+        description: 'Smart Sprinkler Timer',
+        onEvent: tuya.onEventSetTime,
+        configure: tuya.configureMagicPacket,
+        exposes: [
+            e.switch().withEndpoint('switch').withDescription('Switch'),
+            e.enum('status', ea.STATE, ['off', 'auto', 'disabled']).withDescription('Status'),
+            e.numeric('countdown', ea.STATE_SET).withUnit('min').withValueMin(1).withValueMax(240).withDescription('Countdown'),
+            e.numeric('countdown_left', ea.STATE).withUnit('min').withValueMin(1).withValueMax(240).withDescription('Countdown Left'),
+            e.numeric('water_current', ea.STATE).withUnit('L/min').withValueMin(0).withValueMax(3785.41).withValueStep(0.001)
+                .withDescription('Current Water Flow (L/min)'),
+            e.numeric('battery_percentage', ea.STATE).withUnit('%').withValueMin(0).withValueMax(100).withDescription('Battery Percentage'),
+            e.numeric('water_total', ea.STATE).withUnit('L').withValueMin(0).withValueMax(378541.0).withValueStep(0.001)
+                .withDescription('Total Water Flow (L)'),
+            e.enum('weather_delay', ea.STATE_SET, ['24h', '48h', '72h', 'cancel']).withDescription('Weather Delay'),
+            e.text('normal_timer', ea.STATE_SET).withDescription('Normal Timer'),
+            e.binary('weather_switch', ea.STATE_SET, 'ON', 'OFF').withDescription('Weather Switch'),
+            e.binary('switch_enabled', ea.STATE_SET, 'ON', 'OFF').withDescription('Switch Enabled'),
+            e.numeric('surplus_flow', ea.STATE).withDescription('Surplus Flow'),
+            e.numeric('single_watering_duration', ea.STATE).withDescription('Single Watering Duration'),
+            e.numeric('interface_refresh', ea.STATE).withDescription('Interface Refresh'),
```

Comment:
Could you improve the descriptions? E.g. for this one it's quite unclear what it does.

Reply:
Done